ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1603,3 +1612,361 @@ def test_append_arguments_to_list_sources(self):
+            assert check["kw"]["msg"] == error_msg
+            assert "marked as not trusted" in check["kw"]["reason"]
+            assert check["kw"]["key"] == "%s:Server-Cert" % instance_dir
+
+    @pytest.fixture
+    def rename_raagent_cert(self):
+        """
+        Fixture to rename IPA RA CRT and revert
+        """
+        self.master.run_command(
+            ["mv", paths.RA_AGENT_PEM, "%s.old" % paths.RA_AGENT_PEM]
+        )
+        yield
+        self.master.run_command(
+            ["mv", "%s.old" % paths.RA_AGENT_PEM, paths.RA_AGENT_PEM]
+        )
+
+    def test_ipahealthcheck_iparaagent(self, rename_raagent_cert):
+        """
+        Testcase checks that ERROR message is displayed
+        when IPA RA crt file is renamed
+        """
+        error_msg = (
+            "[Errno 2] No such file or directory: '{}'"
+            .format(paths.RA_AGENT_PEM)
+        )
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPARAAgent"
+        )
+        assert returncode == 1
+        for check in data:
+            assert check["result"] == "ERROR"
+            assert check["kw"]["error"] == error_msg
+
+    @pytest.fixture
+    def update_ra_cert_desc(self):
+        """
+        Fixture to modify description of RA cert in ldap
+        and revert
+        """
+        ldap = self.master.ldap_connect()
+        dn = DN(("uid", "ipara"), ("ou", "People"), ("o", "ipaca"))
+        entry = ldap.get_entry(dn)  # pylint: disable=no-member
+        ldap_cert_desc = entry.single_value.get("description")
+
+        def _update_entry(description):
+            entry = ldap.get_entry(dn)  # pylint: disable=no-member
+            entry.single_value['description'] = description
+            ldap.update_entry(entry)  # pylint: disable=no-member
+
+        yield _update_entry
+
+        entry = ldap.get_entry(dn)  # pylint: disable=no-member
+        entry.single_value['description'] = ldap_cert_desc
+        ldap.update_entry(entry)  # pylint: disable=no-member
+
+    def test_ipahealthcheck_iparaagent_ldap(self, update_ra_cert_desc):
+        """
+        Test to check that when description of RA cert in ldap
+        is modified, healthcheck tool reports the correct message
+        """
+        error_msg = 'RA agent not found in LDAP'
+        update_ra_cert_desc('200,12,CN=abc,CN=test')
+        returncode, data = run_healthcheck(
+            self.master,
+            "ipahealthcheck.ipa.certs",
+            "IPARAAgent",
+        )
+        assert returncode == 1
+        for check in data:
+            assert check["result"] == "ERROR"
+            assert check["kw"]["msg"] == error_msg
+
+    def test_ipahealthcheck_iparaagent_bad_serial(self, update_ra_cert_desc):
+        """
+        Test to check cert description doesnt match the expected
+        """
+        error_msg = 'RA agent description does not match. Found {got} ' \
+                    'in LDAP and expected {expected}'
```

Comment:
@rcritten another example of variables not substituted in the error message.

Reply:
Same, it's expected.